lib/Dialect/HW/InferIntRangeInterfaceImpls.cpp

```diff
@@ -0,0 +1,30 @@
+//===- InferIntRangeInterfaceImpls.cpp - Integer range impls for arith -===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#include "circt/Dialect/HW/HWOps.h"
+
+#include "mlir/Interfaces/InferIntRangeInterface.h"
+#include "mlir/Interfaces/Utils/InferIntRangeCommon.h"
+
+#include "llvm/Support/Debug.h"
+#include <optional>
+
+#define DEBUG_TYPE "int-range-analysis"
+
+using namespace mlir;
+using namespace mlir::intrange;
+using namespace circt;
+using namespace circt::hw;
```

Comment:
Is this necessary? Doesn't look to me like you ever implicitly refer to anything in `hw`